### ISSUE TITLE
Tab opening loop

### ISSUE THREAD

**Issue by mildred** [stoically/temporary-containers#311]
- Temporary Containers Version: 1.3
- Firefox Version: 69.0.1 (64-bit)
- Operating System + Version: GNU/Linux Fedora 30
- Non-default Options/Preferences in Temporary Containers:

    ```json
    {
      "version": "1.3",
      "date": 1570623031688,
      "preferences": {
        "automaticMode": {
          "active": true,
          "newTab": "created"
        },
        "notifications": false,
        "container": {
          "namePrefix": "c",
          "color": "pink",
          "colorRandom": false,
          "icon": "circle",
          "iconRandom": false,
          "numberMode": "reuse",
          "removal": "15minutes",
          "colorRandomExcluded": [],
          "iconRandomExcluded": []
        },
        "iconColor": "default",
        "isolation": {
          "global": {
            "navigation": {
              "action": "never"
            },
            "mouseClick": {
              "middle": {
                "action": "notsamedomainexact",
                "container": "default"
              },
              "ctrlleft": {
                "action": "notsamedomainexact",
                "container": "default"
              },
              "left": {
                "action": "never",
                "container": "default"
              }
            },
            "excluded": {},
            "excludedContainers": []
          },
          "domain": [],
          "mac": {
            "action": "disabled"
          },
          "active": true
        },
        "browserActionPopup": true,
        "pageAction": false,
        "contextMenu": true,
        "keyboardShortcuts": {
          "AltC": true,
          "AltP": true,
          "AltN": false,
          "AltShiftC": false,
          "AltX": false,
          "AltO": false
        },
        "replaceTabs": false,
        "cookies": {
          "domain": {}
        },
        "deletesHistory": {
          "automaticMode": "never",
          "contextMenu": false,
          "containerAlwaysPerDomain": "never",
          "containerIsolation": "never",
          "containerRemoval": "instant",
          "containerMouseClicks": "never",
          "statistics": false,
          "active": false,
          "contextMenuBookmarks": false
        },
        "statistics": false,
        "closeRedirectorTabs": {
          "active": false,
          "delay": 2000,
          "domains": [
            "t.co",
            "outgoing.prod.mozaws.net",
            "slack-redir.net"
          ]
        },
        "contextMenuBookmarks": false,
        "ignoreRequests": [
          "getpocket.com",
          "addons.mozilla.org"
        ],
        "ui": {
          "expandPreferences": false,
          "popupDefaultTab": "isolation-per-domain"
        }
      }
    }
    ```

- Other installed Add-ons + Version:
    - Amazon.co.uk	1.1
    - Bing	1.0
    - Block Image|Video	0.1.1
    - DuckDuckGo	1.0
    - Firefox Multi-Account Containers	6.1.1
    - Google	1.0
    - Intégration à GNOME Shell	10.1
    - Javascript Control	1.1
    - PassFF	1.9
    - QRScanner	1.2.3
    - SixOrNot	3.1.2
    - Switch Container	0.8
    - Temporary Containers	1.3
    - Twitter	1.0
    - uBlock Origin	1.22.4
    - Wikipédia (fr)	1.0

### Actual behavior

Sometimes, and it's hard to reproduce, I have firefox that goes crazy and new tabs are constantly opened, one after another (without the old tab being closed). It doesn't stop. If I focus on another window, the tabs are opened in this window instead. The only way to stop it is to close the windows that got infected (not necessarily quit firefox).

It seems to come from a specific tab. Last time I had this problem was a tab pinned to a specific container where I just had clicked (always open in this container) checkbox. However, since then I removed this setting from the multi-account container addon but the problem still persist.

It seems to infect the focused window instead of the window where the tab is located. But as long as the tab exists, the opening loop continues

I believe this is a conflict between temporary containers and multi-account containers where each one is trying to switch the container of a given tab in a loop. If I disable temporary containers, I don't have te problem.

### Expected behavior

No tab opening loop

### Steps to reproduce

Hard to define

### Debug log

I'll save this issue before trying to get a debug log.

edit: unfortunately, I couldn't reproduce the issue. If I encounter it again, and if I can reproduce it, I'll include a debug log.

**Comment by taladar** [comment]
I have the same problem. It seems to occur quite often and it can be fixed without closing the affected window when browser.tabs.closeWindowWithLastTab is disabled and I close the tabs quickly via keyboard shortcuts. There seems to be some sort of small window of opportunity where a tab can be closed before it triggers the creation of another tab.

I could probably get a debug log of the issue if someone described how to enable a relevant debug log. All the search results are just full of the web developer tools instead of ways to debug Firefox behaviour itself.

**Comment by stoically** [comment]
A debug log would be really helpful since I'm unfortunately not able to reproduce. Instructions to get one are [here](https://github.com/stoically/temporary-containers/wiki/Debug-Log).

**Comment by mildred** [comment]
I too have `browser.tabs.closeWindowWithLastTab` disabled.

I finally have debug logs... https://gist.github.com/mildred/35190366257f8bc4b4459c14bde64590

When I activated the logs, I started having `[removeTab] queuing removal of tab to prevent closing of window` messages in a loop, even before I opened the window with the problem. I then opened a new window, navigated to the page that caused the issue, and started having new tabs piling up. I waited a little while. I then closed the window, then stopped the logs.

**Comment by taladar** [comment]
I tried to run the browser for a day with debug logging enabled but couldn't reproduce the problem while it was on. It did, however, happen a minute before I started debug logging and a few minutes after I stopped, so I assume it is a timing issue and the behaviour actually changed due to the changed timing when debug logging.

I also did some general cleanup of history, old cookies and storage and web workers and some vacuum and analyze in the Firefox sqlite files and now the problem happens a lot faster than before when it occurs (new tabs at a rate of many a second instead of one every second or two).

**Comment by stoically** [comment]
@mildred Thanks for the log. Unfortunately it only shows that new tabs are opened, but no reason *why* they're opened - this is *really* odd. Hopefully I have the capacity soon to try debugging this.

@taladar That's really odd as well, thanks for the hint. So maybe I have to artificially slow some stuff down. Do you by chance see the `queuing removal` log entries like @mildred as well?

**Comment by stoically** [comment]
Oh and btw, if you get the chance to test this with a new profile and without other add-ons, that'd be helpful as well. Just to make sure it isn't an add-on incompatibility or firefox configuration somehow.

**Comment by stoically** [comment]
Found the reason for the `queuing removal` spam in the logs, that one is easy to solve, but it unfortunately has nothing to do with the tab opening loop.

I can't reproduce the loop, also not with logging enabled. So if anyone experiences this and can provide another [debug log](https://github.com/stoically/temporary-containers/wiki/Debug-Log), that'd be great. Thanks.

**Comment by stoically** [comment]
So I've pushed out a fix regarding `about:blank` tabs and I think it might've something to do with this tab loop. If you have a chance to test [v1.4beta2](https://github.com/stoically/temporary-containers/releases) and see if the problem still shows up, that'd be helpful, thanks.

**Comment by stoically** [comment]
Published as v1.4. Let me know if the problem pops up again.
